snuba/subscriptions/worker.py

```diff
@@ -109,23 +117,80 @@ def __execute(
-            # not particularly useful in this context and duplicates data that
-            # is already being published to the query log.
-            # XXX: The ``request`` instance is copied when passed to
-            # ``parse_and_run_query`` since it can/will be mutated during
-            # processing.
-            return (
-                request,
-                parse_and_run_query(
+            is_consistent_query = request.settings.get_consistent()
+
+            def run_consistent() -> Result:
+                request_copy = copy.deepcopy(request)
+                request_copy.settings.set_consistent(True)
+
+                return parse_and_run_query(
+                    self.__dataset,
+                    request_copy,
+                    timer,
+                    robust=True,
+                    concurrent_queries_gauge=self.__concurrent_clickhouse_gauge,
+                ).result
+
+            def run_non_consistent() -> Result:
+                request_copy = copy.deepcopy(request)
+                request_copy.settings.set_consistent(False)
+
+                return parse_and_run_query(
                     self.__dataset,
                     copy.deepcopy(request),
                     timer,
                     robust=True,
                     concurrent_queries_gauge=self.__concurrent_clickhouse_gauge,
-                ).result,
+                ).result
+
+            def selector_func(run_snuplicator: bool) -> Tuple[str, List[str]]:
+                primary = "consistent" if is_consistent_query else "non_consistent"
+                other = "non_consistent" if is_consistent_query else "consistent"
+
+                return (primary, [other] if run_snuplicator else [])
+
+            def callback_func(
+                data: List[ThreadedFunctionDelegatorResult[Result]],
+            ) -> None:
+                primary_result = data.pop(0).result
+
+                for result in data:
+                    match = result.result == primary_result
+                    self.__metrics.increment(
+                        "consistent",
+                        tags={
+                            "dataset": self.__dataset_name,
+                            "consistent": str(is_consistent_query),
+                            "match": str(match),
+                        },
+                    )
+
+            if self.__dataset_name == "events":
+                sample_rate = state.get_config(
+                    "event_subscription_non_consistent_sample_rate", 0
+                )
+            elif self.__dataset_name == "transactions":
+                sample_rate = state.get_config(
+                    "transaction_subscription_non_consistent_sample_rate", 0
+                )
+            else:
+                sample_rate = 0
+
+            assert sample_rate is not None
+
+            run_snuplicator = random.random() < float(sample_rate)
```

Comment:
Could you please set `run_snuplicator` = False if the request comes in at this point with consistency = False?
We have already the consistency bypass in place to shed load if needed. So if that is in use this could potentially increase the amount of consistent queries. 
I would run the experiment only for queries that have `consistent` = True to begin with, so the queries we add can only be non consistent.

Reply:
I would prefer not to try to be too clever about this here. If we need to shed load I think we should just set these values to 0 and get the same result rather than have various options influence each other in a way that may be unexpected. It means that there will an extra option to update though - is the concern that updating in this extra place will be too much work?

Reply:
I was concerned about forgetting this on. But after seeing the metrics today with the jitter, I think we are fine. You can ignore this comment. Just please see the one on the requestSettings immutability